src/sentry/static/sentry/app/components/charts/baseChart.tsx

```diff
@@ -336,6 +340,14 @@ class BaseChart extends React.Component<Props> {
         )
       : [XAxis(), XAxis()];
 
+    // @ts-ignore Maybe changing the series type to types/echarts Series[] would be a better solution
+    // and can't use ignore for multiline blocks
+    const seriesValid = series.length && series[0].data && series[0].data.length > 1;
+    // @ts-ignore
+    const seriesData = seriesValid ? series[0].data : undefined;
+    // @ts-ignore
+    const bucketSize = seriesData ? seriesData[1][0] - seriesData[0][0] : undefined;
```

Comment:
Generally the chart creator knows the timeseries bucket size as it is is in the request parameters when fetching results. Should this component also accept the `interval` prop used elsewhere?

Reply:
Interval isn't always present in the chart creator components, I would need to just copy paste this code that gets interval/bucketSize from series in most of them, which isn't any better I think.

Reply:
Fair point.